src/struct.h

```diff
@@ -2,6 +2,7 @@
 
 #include <map>
 #include <memory>
+#include <unordered_set>
```

Comment:
is this necessary here or in utils.h/cpp?

Reply:
It's completely superfluous here, removed.